apis/Google.Cloud.Spanner.Data/docs/migration-to-2.md

```diff
@@ -0,0 +1,103 @@
+are used. For users who are affected, this document should help you
+to migrate your application to the 2.0 version of the libraries.
+
+If you have any problems with the breaking changes and they're not
+covered in this document, please [file an
+issue](https://github.com/googleapis/google-cloud-dotnet/issues/new)
+so we can help you further.
+
+## Configuration changes
+
+- `SpannerOptions` has been removed completely. See the [configuration
+  guide](configuration.md) for more details, but for migration
+  purposes:
+  - `KeepAliveInterval` is replaced by `SessionPoolOptions.IdleSessionRefreshDelay`
+  - `MaximumPooledSessions` has no direct equivalent in 2.0; once a session
+  is part of the pool, it will be refreshed as necessary until it is evicted.
+  - `MaximumActiveSessions` is now only present in `SessionPoolOptions`
+  - `MaximumGrpcChannels` is specified in the connection string
+  - `LogLevel` is controlled on a per-logger basis; this property on
+    `SpannerOptions` was equivalent to changing
+    `Logger.DefaultLogger.LogLevel`.
+  - `PoolEvictionDelay` is now only present in `SessionPoolOptions`
+  - `ResourcesExhaustedBehavior` has moved to `SessionPoolOptions`
+  - `Timeout` is specified in the connection string
+  - `MaximumConcurrentSessionCreates` is now only present in
+    `SessionPoolOptions`
+- `ClientPool` has been removed in 2.0. Its functionality has largely
+  been replaced by `SessionPoolManager`. It is expected that the
+  default session pool manager will be appropriate for most use cases,
+  but you can associate a `SessionPoolManager` with the
+  `SpannerConnectionStringBuilder` used to build a connection for more
+  advanced situations.
+- `ResourcesExhaustedBehavior` has moved to Google.Cloud.Spanner.V1,
+  as it is now configured in `SessionPoolOptions`.
+- Loggers are configured on a `SessionPoolManager` basis
+
+## Breaking changes in connection, command and transaction types
+
+- The `SpannerCommandTextBuilder(string)` constructor has been removed.
+  The equivalent functionality is available via the static
+  `SpannerCommandTextBuilder.FromCommandText` method.
+- `SpannerTransaction.CommitAsync` returns a `Task<DateTime>` rather
+  than a `Task<DateTime?>`. The server will always return a commit
+  timestamp, so the task's result would never have been null in 1.0
+  anyway. Client code should be able to use the resulting
+  `Task<DateTime>` more easily in most cases.
+- `SpannerConnection` and `SpannerConnectionStringBuilder` no longer
+  expose `GetCredentials` method. There is no direct equivalent in 2.0;
+  please file an issue if you rely on this functionality.
+- `SpannerConnection.ClearPooledResourcesAsync` is replaced by
+  `SpannerConnection.ShutdownSessionPoolAsync`. After this is called,
+  no further sessions can be created with the associated session pool.
+  If you need to control sessions in a more fine-grained way,
+  you can associate a `SessionPoolManager` with the
```

Comment:
Does this implies that they should inherit from SessionPoolManager? Because whatever the options on the session pool, shutting down is the same. Basically what do you mean with "control sessions in a more fine-grained way"?

Reply:
I've tried to clarify this a bit. See what you think.